src/mscorlib/src/System/Globalization/EncodingTable.Unix.cs

```diff
@@ -1,101 +1,162 @@
+            //If we didn't find it, try looking it up now.
+            //If we find it, add it to the hashtable.
+            //This is a linear search, but we probably won't be doing it very often.
+            for (int i = 0; i < encodingDataTableItems.Length; i++)
+            {
+                CodePageDataItem temp = encodingDataTableItems[i];
+                if (temp.CodePage == codepage)
+                {
+                    dataItem = temp;
+                    hashByCodePage[codepage] = dataItem;
+                    break;
+                }
             }
+
+            return dataItem;
         }
 
         // PAL ends here.
 
-        const int CodePageUtf7 = 65000;
-        const int CodePageUtf8 = 65001;
-        const int CodePageUtf16 = 1200;
-        const int CodePageUtf16BE = 1201;
-        const int CodePageUtf32 = 12000;
+        private static Dictionary<string, ushort> encodingDataTable = 
+            new Dictionary<string, ushort>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "ANSI_X3.4-1968", 20127 },
+            { "ANSI_X3.4-1986", 20127 },
+            { "ascii", 20127 },
+            { "cp367", 20127 },
+            { "cp819", 28591 },
+            { "csASCII", 20127 },
+            { "csISOLatin1", 28591 },
+            { "csUnicode11UTF7", 65000 },
+            { "IBM367", 20127 },
+            { "ibm819", 28591 },
+            { "ISO-10646-UCS-2", 1200 },
+            { "iso-8859-1", 28591 },
+            { "iso-ir-100", 28591 },
+            { "iso-ir-6", 20127 },
+            { "ISO646-US", 20127 },
+            { "iso8859-1", 28591 },
+            { "ISO_646.irv:1991", 20127 },
+            { "iso_8859-1", 28591 },
+            { "iso_8859-1:1987", 28591 },
+            { "l1", 28591 },
+            { "latin1", 28591 },
+            { "ucs-2", 1200 },
+            { "unicode", 1200},
+            { "unicode-1-1-utf-7", 65000 },
+            { "unicode-1-1-utf-8", 65001 },
+            { "unicode-2-0-utf-7", 65000 },
+            { "unicode-2-0-utf-8", 65001 },
+            // People get confused about the FFFE here.  We can't change this because it'd break existing apps.
+            // This has been this way for a long time, including in Mlang.
+            // Big Endian, BOM seems backwards, think of the BOM in little endian order.
+            { "unicodeFFFE", 1201},
+            { "us", 20127 },
+            { "us-ascii", 20127 },
+            { "utf-16", 1200 },
+            { "UTF-16BE", 1201},
+            { "UTF-16LE", 1200},
+            { "utf-32", 12000 },
+            { "UTF-32BE", 12001 },
+            { "UTF-32LE", 12000 },
+            { "utf-7", 65000 },
+            { "utf-8", 65001 },
+            { "x-unicode-1-1-utf-7", 65000 },
+            { "x-unicode-1-1-utf-8", 65001 },
+            { "x-unicode-2-0-utf-7", 65000 },
+            { "x-unicode-2-0-utf-8", 65001 },
+        };
```

Comment:
Where does this list of pairs come from?  Would be worth a comment I think explaining how it was derived.